app/containers/HomePage/HomePage.react.js

```diff
@@ -6,61 +6,53 @@
 import { asyncChangeProjectName, asyncChangeOwnerName } from './HomePage.actions';
 import React from 'react';
 import { connect } from 'react-redux';
+import { routeActions } from 'redux-simple-router';
 
 import Button from 'Button/Button.react';
 
 import styles from './HomePage.css';
 
-class HomePage extends React.Component {
-  constructor() {
-    super();
-    this.onChangeProjectName = this.onChangeProjectName.bind(this);
-    this.onChangeOwnerName = this.onChangeOwnerName.bind(this);
-  }
-
-  onChangeProjectName(evt) {
-    this.props.dispatch(asyncChangeProjectName(evt.target.value));
-  }
-
-  onChangeOwnerName(evt) {
-    this.props.dispatch(asyncChangeOwnerName(evt.target.value));
-  }
-
-  render() {
-    const { projectName, ownerName } = this.props.data;
-
-    return (
-      <div>
-        <h1>Hello World!</h1>
-        <h2>This is the demo for the { projectName } by
-          <a href={`https://twitter.com/${ownerName}`} className={styles.link} > @{ ownerName }</a>
-        </h2>
-        <label className={styles.label}>Change to your project name:
-          <input className={styles.input} type="text"
-            onChange={this.onChangeProjectName}
-            defaultValue="React.js Boilerplate" value={projectName}
-          />
-        </label>
-        <label className={styles.label}>Change to your name:
-          <input className={styles.input} type="text"
-            onChange={this.onChangeOwnerName}
-            defaultValue="mxstbr" value={ownerName}
-          />
-        </label>
-        <Button route="/readme">Setup</Button>
-      </div>
-    );
-  }
+function HomePage(props) {
+  const { projectName, ownerName } = props.data;
+  const { pathname } = props.loc;
+  return (
+    <div>
+      <h1>Hello World!</h1>
+      <h2>This is the demo for the { projectName } by
+        <a href={`https://twitter.com/${ownerName}`} className={styles.link} > @{ ownerName }</a>
+      </h2>
+      <label className={styles.label}>Change to your project name:
+        <input className={styles.input} type="text"
+          onChange={ (evt) => { props.onChangeProjectName(evt.target.value); }}
+          defaultValue="React.js Boilerplate" value={projectName}
+        />
+      </label>
+      <label className={styles.label}>Change to your name:
+        <input className={styles.input} type="text"
+          onChange={ (evt) => props.onChangeOwnerName(evt.target.value) }
+          defaultValue="mxstbr" value={ownerName}
+        />
+      </label>
+      <Button handleRoute= { () => props.onChangeRoute('/readme') } >Setup</Button>
+      <p> Here is {"'"}{ pathname }{"'"}</p>
+    </div>
+  );
 }
 
-// REDUX STUFF
+function mapStateToProps(state) {
+  return {
+    loc: state.routing.location,
```

Comment:
Okay , I'll rename these. BTW , The reason why I used shortening names is to give me a hint that `loc` is merely a meaningless variable rather a specific property.